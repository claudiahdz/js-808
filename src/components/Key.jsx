import React, { Component } from 'react';

class Key extends Component {

  state = {
    active: false,
  }

  componentWillMount = () => {
    this.init(this.props)
  }

  componentWillReceiveProps = (nextProps) => {
    this.init(nextProps)
  }

  init = (props) => {
    this.setState({
      active: props.active || false,
    })
  }

  onPressKey = () => {
    this.setState({ active: !this.state.active })
  }

  render() {
    const { active } = this.state

    return (
      <div className={`key ${active && 'active'}`} onClick={this.onPressKey}></div>
    )
  }
}

export default Key;