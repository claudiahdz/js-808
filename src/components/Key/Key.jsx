import React from 'react'
import PropTypes from 'prop-types'

const Key = ({ active, updateBeat, row, column }) => {
  return (
    <div 
      className={active ? 'key active' : 'key'} 
      onClick={() => updateBeat(row, column)}
    >
    </div>
  )
}

Key.propTypes = {
  active: PropTypes.number.isRequired,
  updateBeat: PropTypes.func.isRequired,
  row: PropTypes.number.isRequired,
  column: PropTypes.number.isRequired
}

export default Key