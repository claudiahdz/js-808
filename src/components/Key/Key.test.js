import React from 'react'
import { shallow } from 'enzyme'
import Key from './Key'

describe('Key', () => {
  const defaultProps = {
    active: 0,
    updateBeat: () => {},
    row: 0,
    column: 0,
  }

  it('renders the component correctly', () => {
    const key = shallow(<Key {...defaultProps} />)
    expect(key).toMatchSnapshot()
  })

  it('calls updateBeat when clicked', () => {
    const props = {
      ...defaultProps,
      updateBeat: jest.fn()
    }
    const key = shallow(<Key {...props} />)
    key.simulate('click')
    
    expect(props.updateBeat).toHaveBeenCalledWith(props.row, props.column)
  })
})

