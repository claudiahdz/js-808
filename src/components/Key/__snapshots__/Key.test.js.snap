// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key renders the component correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="key 0"
    onClick={[Function]}
/>,
  "nodes": Array [
    <div
      className="key 0"
      onClick={[Function]}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Key
        active={0}
        column={0}
        row={0}
        updateBeat={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "active": 0,
          "column": 0,
          "row": 0,
          "updateBeat": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="key 0"
          onClick={[Function]}
/>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="key 0"
          onClick={[Function]}
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Key
    active={0}
    column={0}
    row={0}
    updateBeat={[Function]}
/>,
}
`;
